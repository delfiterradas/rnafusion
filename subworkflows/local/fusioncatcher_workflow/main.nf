include { FASTP as FASTP_FOR_FUSIONCATCHER } from '../../../modules/nf-core/fastp/main'
include { FUSIONCATCHER_FUSIONCATCHER }      from '../../../modules/nf-core/fusioncatcher/fusioncatcher/main'

// TODO: Remove fusioncatcher_fusions as parameter.
// TODO: remove dummy file. Work with Channel.empty()
// TODO: if the files were already produced and the user want to skip the module because of this, they should be taken them from the sample sheet

workflow FUSIONCATCHER_WORKFLOW {
    take:
        reads                   // channel [ meta, [ fastqs ] ]
        fusioncatcher_trimming     // boolean
        adapter_fasta         // channel [ path ]
        fusioncatcher_ref       // channel [ meta, path       ]
        fusioncatcher_fusions   // path, string

    main:
        ch_versions   = Channel.empty()

        if (fusioncatcher_fusions){

            ch_fusioncatcher_fusions = reads.combine(Channel.value(file(fusioncatcher_fusions, checkIfExists:true)))
                                        .map { meta, _reads, fusions -> [ meta, fusions ] }
        } else {
            if (fusioncatcher_trimming) {
                FASTP_FOR_FUSIONCATCHER(
                    reads,
                    adapter_fasta,
                    false, // discard_trimmed_pass
                    false, // save_trimmed_fail
                    false  // skip_qc
                )
                ch_versions = ch_versions.mix(FASTP_FOR_FUSIONCATCHER.out.versions)
                reads = FASTP_FOR_FUSIONCATCHER.out.reads
            }

            FUSIONCATCHER_FUSIONCATCHER (
                reads,
                fusioncatcher_ref
            )
            ch_fusioncatcher_fusions = FUSIONCATCHER_FUSIONCATCHER.out.fusions
            ch_versions              = ch_versions.mix(FUSIONCATCHER_FUSIONCATCHER.out.versions)
        }

    emit:
        fusions  = ch_fusioncatcher_fusions     // channel [ meta, fusions ]
        versions = ch_versions                  // channel [ versions      ]
    }

