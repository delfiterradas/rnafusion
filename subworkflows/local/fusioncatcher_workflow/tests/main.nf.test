nextflow_workflow {

    name "Test Subworkflow FUSIONCATCHER_WORKFLOW"
    script "../main.nf"
    workflow "FUSIONCATCHER_WORKFLOW"
    tag "subworkflow"
    tag "fusioncatcher"
    tag "fusioncatcher/build"
    tag "fusioncatcher/detect"

    // Test
    test("FUSIONCATCHER_WORKFLOW - Homo sapiens - FASTQs chr4") {

        setup {
            // Download reference files for fusioncatch
            run("FUSIONCATCHER_BUILD") {
                script "../../../../modules/local/fusioncatcher/build/main.nf"
                process {
                """
                input[0] = Channel.value('46')
                """
                }
            }
        }

        // TODO: get smaller reference files for fusioncatcher
        when {
            workflow {
                """
                // ch_reads
                input[0] = Channel.of(
                    [
                        [ id: "test_fastqs", single_end: false ],
                        [
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/human/reads_1.fq.gz", checkIfExists: true),
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/human/reads_2.fq.gz", checkIfExists: true)
                        ]
                    ] )

                // ch_references
                input[1] = FUSIONCATCHER_BUILD.out.reference

                // fusioncatcher (boolean)
                input[2] = true

                // all (boolean)
                input[3] = true

                // fusioninspector_only (boolean)
                input[4] = false

                // fusioncatcher_fusions (string path)
                input[5] = null
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

}
