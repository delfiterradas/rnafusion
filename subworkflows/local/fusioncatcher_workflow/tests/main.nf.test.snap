{
    "FUSIONCATCHER_WORKFLOW - Homo sapiens - FASTQs chr4": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_fastqs",
                            "single_end": false
                        },
                        "test_fastqs.fusioncatcher.fusion-genes.txt:md5,c826a24c49abfcec8164c478e1e74892"
                    ]
                ],
                "1": [
                    "versions.yml:md5,05bd93a243728a293211ce52e5f97282"
                ],
                "fusions": [
                    [
                        {
                            "id": "test_fastqs",
                            "single_end": false
                        },
                        "test_fastqs.fusioncatcher.fusion-genes.txt:md5,c826a24c49abfcec8164c478e1e74892"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,05bd93a243728a293211ce52e5f97282"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-03T19:29:54.767628"
    }
}